samtranslator/model/api/api_generator.py

```diff
@@ -576,7 +598,40 @@ def _construct_api_domain(  # noqa: too-many-branches
 
             record_set_group.RecordSets += self._construct_record_sets_for_domain(self.domain, api_domain_name, route53)
 
-        return domain, basepath_resource_list, record_set_group
+        return ApiDomainResponse(domain, basepath_resource_list, record_set_group, None)
+
+    def _construct_individual_record_set(
+        self, domain: Dict[str, Any], api_domain_name: str, route53: Any
+    ) -> List[Route53RecordSet]:
+        hostedZoneId = route53.get("HostedZoneId")
+        hostedZoneName = route53.get("HostedZoneName")
+        domainName = domain.get("DomainName")
+        logical_id_suffix = LogicalIdGenerator("", [hostedZoneId or hostedZoneName, domainName]).gen()
+        logical_id = "RecordSet" + logical_id_suffix
+        logical_id_ipv6 = "RecordSetIpv6" + logical_id_suffix
+
+        recordset_list = []
+
+        recordset = Route53RecordSet(logical_id, attributes=self.passthrough_resource_attributes)
+        recordset.Name = domain.get("DomainName")
+        recordset.Type = "A"
+        recordset.AliasTarget = self._construct_alias_target(domain, api_domain_name, route53)
```

Comment:
Can reuse the target for IPv6 record as well.

Reply:
Updated